src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -125,6 +133,312 @@ public GermplasmServiceImpl(final HibernateSessionProvider sessionProvider) {
 		return results;
 	}
 
+	@Override
+	public Set<Integer> importGermplasmUpdates(final List<GermplasmUpdateDTO> germplasmUpdateDTOList) {
+
+		final Multimap<String, Object[]> conflictErrors = ArrayListMultimap.create();
+
+		// Get the names as well as the codes specified in the preferred name property.
+		final Set<String> namesCode = new HashSet<>();
```

Comment:
Can we extract the code to form the maps `nameCodesFieldNoMap` and `attributeCodesFieldNoMap` to separate methods similar to `getNameTypesMapByName` and `getAttributesMapByName`?